src/imap/select_folder.rs

```diff
@@ -205,8 +211,8 @@ impl ImapSession {
                             context,
                             "The server illegally decreased the uid_next of folder {folder:?} from {old_uid_next} to {new_uid_next} without changing validity ({new_uid_validity}), resyncing UIDs...",
                         );
-                        set_uid_next(context, folder, new_uid_next).await?;
```

Comment:
Why reordering this? Shouldn't a resync be requested after we successfully set the new UIDNEXT?